src/parser/mod.rs

```diff
@@ -5470,6 +5463,13 @@ impl<'a> Parser<'a> {
 
         let strict = self.parse_keyword(Keyword::STRICT);
 
+        // Parse optional `AS ( query )`
+        let query = if self.parse_keyword(Keyword::AS) {
+            Some(self.parse_boxed_query()?)
+        } else {
+            None
+        };
+
```

Comment:
Should the block come after the `COMMENT` is parsed? (not sure re which dialects use the comment syntax  and I'm assuming the `AS` is usually the last word in the create table).

quick look at the display function for create table seemed like we print the comment much earlier in the statement which might be weird (unrelated to this change I imagine though, just noticed while looking there)

Reply:
@iffyio Yes, good call. The comment syntax was introduced in PR https://github.com/sqlparser-rs/sqlparser-rs/pull/1059 for MySQL. And after looking into the MySQL syntax, it's included in the table option. So you're right that it should parsed after the AS query.

```MySQL
table_option: {
    AUTOEXTEND_SIZE [=] value
  | AUTO_INCREMENT [=] value
  | AVG_ROW_LENGTH [=] value
  | [DEFAULT] CHARACTER SET [=] charset_name
  | CHECKSUM [=] {0 | 1}
  | [DEFAULT] COLLATE [=] collation_name
  | COMMENT [=] 'string']
 ...
```

refer to: https://dev.mysql.com/doc/refman/8.4/en/create-table.html